### PR TITLE
HPCC-12153 ECL Watch Security Improvements

### PR DESCRIPTION
**Pull request by miguelvazq**
Fixes: Permissions advanced menu items always active (HPCC-11533)
Fixes: Adding/Removing user to/from group is not clear (HPCC-12318)
Fixes: Adds physical file permission check
Fixes: Adds inherited permissions to users
Fixes: Adds available permissions to users & groups
Fixes: Adds clear permissions cache on all appropriate permission areas
Fixes: Adds default file permission
Fixes: Adds check file permission
Fixes: Various U/I improvements

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/ShowAccountPermissionsWidget.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
+/*##############################################################################
+#    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
```
Comment (GordonSmith) [inline esp/src/eclwatch/ShowAccountPermissionsWidget.js]: 2016?

File: esp/src/eclwatch/ShowAccountPermissionsWidget.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    "dgrid/editor",
+
+    "hpcc/GridDetailsWidget",
+    "hpcc/ws_access",
+    "hpcc/ESPUtil",
+    "hpcc/ShowInheritedPermissionsWidget",
+
+    "dijit/layout/BorderContainer",
+    "dijit/layout/TabContainer",
+    "dijit/layout/ContentPane"
```
Comment (GordonSmith) [inline esp/src/eclwatch/ShowAccountPermissionsWidget.js]: Are these three dijits needed?

File: esp/src/eclwatch/ShowIndividualPermissionsWidget.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+    "dgrid/editor",
+
+    "hpcc/GridDetailsWidget",
+    "hpcc/ws_access",
+    "hpcc/ESPUtil",
+
+    "dijit/layout/BorderContainer",
+    "dijit/layout/TabContainer",
+    "dijit/layout/ContentPane"
```
Comment (GordonSmith) [inline esp/src/eclwatch/ShowIndividualPermissionsWidget.js]: Are these 3 dijits needed?  If they are in GridDetailsWidget.js then they won't be...

File: esp/src/eclwatch/UserQueryWidget.js [hunk 4]
```diff
@@ -643,6 +797,71 @@ define([
             }
         },
 
+        ensurePermissionsPane: function (id, params) {
+            id = this.createChildTabID(id);
+            var retVal = registry.byId(id);
+            if (!retVal) {
+                retVal = new ShowIndividualPermissionsWidget({
+                    id: id,
+                    title: params.Name,
```
Comment (GordonSmith) [inline esp/src/eclwatch/UserQueryWidget.js]: Should this be params.TabName?

Comment (miguelvazq) [inline esp/src/eclwatch/UserQueryWidget.js]: I don't pass in TabName in regular ensurePermissionsPane() just in the ones I don't have a scope in.

File: esp/src/eclwatch/UserQueryWidget.js [hunk 5]
```diff
@@ -643,6 +797,71 @@ define([
...
+                    id: id,
+                    title: params.TabName,
+                    iconClass: 'iconPeople',
+                    closable: true,
+                    params: params
+                });
+                this.addChild(retVal, "last");
+            }
+            return retVal;
+        },
```
Comment (GordonSmith) [inline esp/src/eclwatch/UserQueryWidget.js]: Are these 4 ensureXXXPermissionsPane the same?  Could they be commoned up?

File: esp/src/eclwatch/templates/UserQueryWidget.html [hunk 6]
```diff
@@ -127,8 +132,23 @@
...
                 </div>
             </div>
-            <div id="${id}_FileScopes" title="${i18n.FileScopes}" style="padding: 0px; border:0px; border-color:none; overflow: hidden" data-dojo-type="dijit.layout.ContentPane">
-            </div>
         </div>
     </div>
+    <div id="${id}FilePermissionDialog" data-dojo-type="dijit.Dialog" title="${i18n.CheckFilePermissions}">
+        <form id="${id}ZapForm" style="width:460px">
+            <div data-dojo-props="cols:2" data-dojo-type="hpcc.TableContainer">
+                <p>Please select a user or a group along with a file name</p>
```
Comment (GordonSmith) [inline esp/src/eclwatch/templates/UserQueryWidget.html]: Translation String

**Comment by GordonSmith** [comment]
Look ok - there seems to be some code duplication would could possibly be commoned up?

**Comment by miguelvazq** [comment]
@GordonSmith all changes made and tested.

**Comment by GordonSmith** [comment]
Looks ok - some code duplication, but it can be pulled.

**Comment by miguelvazq** [comment]
@GordonSmith code duplication within ws_access? I fixed the ones you spoke about?

**Pull request merged**